benchmark/src/main/java/com/datatorrent/benchmark/monitor/SystemResourceMonitor.java

```diff
@@ -0,0 +1,170 @@
+
+  protected static SystemResourceMonitor defaultInstance;
+
+  public static SystemResourceMonitor create()
+  {
+    if (defaultInstance == null) {
+      synchronized (SystemResourceMonitor.class) {
+        if (defaultInstance == null) {
+          defaultInstance = new SystemResourceMonitor();
+        }
+      }
+    }
+    return defaultInstance;
+  }
+
+  public static SystemResourceMonitor create(SystemResource[] systemResources)
+  {
+    SystemResourceMonitor monitor = resourcesToMonitors.get(Sets.newHashSet(systemResources));
+    if (monitor == null) {
+      synchronized (SystemResourceMonitor.class) {
+        if (monitor == null) {
+          monitor = new SystemResourceMonitor(systemResources);
+          resourcesToMonitors.put(Sets.newHashSet(systemResources), monitor);
+        }
+      }
+    }
+    return monitor;
+  }
+
+  protected SystemResource[] systemResources;
+  protected OperatingSystemMXBean operatingSystemMXBean = ManagementFactory.getOperatingSystemMXBean();
+  protected Map<SystemResource, Number> systemResourceUsage = Maps.newEnumMap(SystemResource.class);
+  protected List<GarbageCollectorMXBean> garbageCollectorMXBeans = ManagementFactory.getGarbageCollectorMXBeans();
+
+  protected SystemResourceMonitor()
+  {
+  }
+
+  protected SystemResourceMonitor(SystemResource[] systemResources)
+  {
+    if (systemResources == null || systemResources.length == 0) {
+      throw new IllegalArgumentException("Resources should not null or empty");
+    }
+    this.systemResources = systemResources;
+  }
+
+  public Map<SystemResource, Number> getSystemResourceUsage()
+  {
+    systemResourceUsage.clear();
+    for (SystemResource sr : systemResources) {
+      Method getterMethod = getGetterMethod(sr);
+      getterMethod.setAccessible(true);
+      try {
+        systemResourceUsage.put(sr, (Number)getterMethod.invoke(operatingSystemMXBean));
+      } catch (Exception e) {
+        throw new RuntimeException("Can't invoke method: " + getterMethod.getName(), e);
+      }
+    }
+    return systemResourceUsage;
+  }
+
+  public String getAppName()
+  {
+    return operatingSystemMXBean.getName();
+  }
+
+  public String getFormattedSystemResourceUsage()
+  {
+    Map<SystemResource, Number> usage = getSystemResourceUsage();
+    StringBuilder sb = new StringBuilder();
+    for (Map.Entry<SystemResource, Number> entry : usage.entrySet()) {
+      formatEntry(entry, sb);
+    }
+    return sb.toString();
+  }
+
+  protected void formatEntry(Map.Entry<SystemResource, Number> entry, StringBuilder sb)
+  {
+    sb.append("\n\t").append(entry.getKey()).append(": ").append(NumberFormat.getInstance().format(entry.getValue()));
+  }
+
+  protected Method getGetterMethod(SystemResource sr)
+  {
+    try {
+      return operatingSystemMXBean.getClass().getMethod("get" + sr.name(), null);
+    } catch (Exception e) {
+      throw new RuntimeException("Invalid SystemResource: " + sr);
```

Comment:
use Throwables.propagate and please also look at other places that does a new RuntimeException